extensions/amp-ad/0.1/test/test-amp-ad-3p-impl.js

```diff
@@ -216,235 +140,257 @@ function tests(name) {
+        expect(ad.implementation_.iframe_).to.be.null;
       });
+    });
 
-
-      it('should hide placeholder when ad falls back', () => {
-        return getAd(undefined, undefined, ad => {
-          const placeholder = document.createElement('div');
-          placeholder.setAttribute('placeholder', '');
-          ad.appendChild(placeholder);
-          expect(placeholder.classList.contains('amp-hidden')).to.be.false;
-
-          const fallback = document.createElement('div');
-          fallback.setAttribute('fallback', '');
-          ad.appendChild(fallback);
-          return ad;
-        }).then(ad => {
-          const placeholderEl = ad.querySelector('[placeholder]');
-          sandbox.stub(
-              ad.implementation_, 'deferMutate', function(callback) {
-                callback();
-              });
-          ad.implementation_.noContentHandler_();
-          expect(placeholderEl.classList.contains('amp-hidden')).to.be.true;
-        });
+    it('should not destroy a master iframe', () => {
+      return getAd().then(ad => {
+        ad.implementation_.iframe_.setAttribute(
+            'name', 'frame_doubleclick_master');
+        sandbox.stub(
+            ad.implementation_, 'deferMutate', function(callback) {
+              callback();
+            });
+        ad.implementation_.noContentHandler_();
+        expect(ad.implementation_.iframe_).to.not.be.null;
       });
-
-      it('should destroy non-master iframe', () => {
-        return getAd(undefined, undefined, ad => {
-          const placeholder = document.createElement('div');
-          placeholder.setAttribute('placeholder', '');
-          ad.appendChild(placeholder);
-          expect(placeholder.classList.contains('amp-hidden')).to.be.false;
-
-          const fallback = document.createElement('div');
-          fallback.setAttribute('fallback', '');
-          ad.appendChild(fallback);
-          return ad;
-        }).then(ad => {
-          ad.implementation_.iframe_.setAttribute(
-              'name', 'frame_doubleclick_0');
-          sandbox.stub(
-              ad.implementation_, 'deferMutate', function(callback) {
-                callback();
-              });
-          ad.implementation_.noContentHandler_();
-          expect(ad.implementation_.iframe_).to.be.null;
-        });
+    });
+  });
+
+  describe('renderOutsideViewport', () => {
+    it('should not return false after scrolling, then false for 1s', () => {
+      let clock;
+      return getAd(undefined, undefined, ad => {
+        clock = lolex.install(ad.implementation_.win);
+        expect(ad.implementation_.renderOutsideViewport()).not.to.be.false;
+        return ad;
+      }).then(ad => {
+        const impl = ad.implementation_;
+        expect(impl.renderOutsideViewport()).to.be.false;
+        clock.tick(900);
+        expect(impl.renderOutsideViewport()).to.be.false;
+        clock.tick(100);
+        expect(impl.renderOutsideViewport()).not.to.be.false;
```

Comment:
it's not in this special case. `#renderOutsideViewport()` return `{boolean|number}`